libs/api/domains/municipalities-financial-aid/src/lib/municipalitiesFinancialAid.resolver.ts

```diff
@@ -0,0 +1,69 @@
+import { Args, Mutation, Query, Resolver } from '@nestjs/graphql'
+import type { User } from '@island.is/auth-nest-tools'
+import {
+  IdsUserGuard,
+  ScopesGuard,
+  CurrentUser,
+} from '@island.is/auth-nest-tools'
+import { UseGuards } from '@nestjs/common'
+
+import { MunicipalitiesFinancialAidService } from './municipalitiesFinancialAid.service'
+import { MunicipalityModel } from './models/municipality.model'
+import { MunicipalityQueryInput } from './models/municipality.input'
+import { SignedUrlModel } from './models/signedUrl.model'
+import { GetSignedUrlInput } from './dto/getSignedUrl.input'
+import { CreateMunicipalitiesApplicationInput } from './dto/createApplication.input'
+import { ApplicationModel } from './models/application.model'
+
+@UseGuards(IdsUserGuard, ScopesGuard)
+@Resolver()
+export class MunicipalitiesFinancialAidResolver {
+  constructor(
+    private municipalitiesFinancialAidService: MunicipalitiesFinancialAidService,
+  ) {}
+
+  @Query(() => String, { nullable: true })
+  async hasUserFinancialAidApplicationForCurrentPeriod(
+    @CurrentUser() user: User,
+  ): Promise<string | null> {
+    return await this.municipalitiesFinancialAidService.municipalitiesFinancialAidCurrentApplication(
+      user,
+    )
+  }
+
+  @Query(() => MunicipalityModel, { nullable: true })
```

Comment:
Do all of these APIs have scope checks and audit so we don't need to configure it here?

Reply:
@eirikurn  I am not quite sure what you mean, could you explain?

Reply:
I mean all of our API calls should check the audience and scope of access tokens as well be logged in an audit log so we trace what users do.

You can see it done in other endpoints using `IdsUserGuard`, `ScopeGuard`, `@Scopes()`, `@Audit()` or `this.auditService.audit()`.

When our GraphQL calls another API like your financial aid API here, ideally the responsibility is on your API to fulfill this, but when we're calling APIs which don't validate our access tokens or don't have an audit, we do it here in the GraphQL API.

Reply:
Yes those things are taken care by our APIs